src/jit/regset.cpp

```diff
@@ -2339,8 +2339,6 @@ regNumber RegSet::rsUnspillOneReg(GenTreePtr tree, regNumber oldReg, KeepReg wil
 //
 TempDsc* RegSet::rsUnspillInPlace(GenTreePtr tree, regNumber oldReg, unsigned regIdx /* =0 */)
 {
-    assert(!isRegPairType(tree->gtType));
```

Comment:
FWIW, the code I've changed appears to be the only part of the RyuJIT backend that observes `isRegPairType`. We could probably go further and `#ifdef` out `isRegPairType` entirely for RyuJIT.